### PR TITLE
Developer guide

### PR DESCRIPTION
**Pull request by BagDownB3nny**
Edited developer guide for Ultron

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -270,58 +270,115 @@ _{Explain here how the data archiving feature will be implemented}_
...
-| `* * *`  | new user                                   | see usage instructions         | refer to instructions when I forget how to use the App                 |
-| `* * *`  | user                                       | add a new person               |                                                                        |
-| `* * *`  | user                                       | delete a person                | remove entries that I no longer need                                   |
-| `* * *`  | user                                       | find a person by name          | locate details of persons without having to go through the entire list |
-| `* *`    | user                                       | hide private contact details   | minimize chance of someone else seeing them by accident                |
-| `*`      | user with many persons in the address book | sort persons by name           | locate a person easily                                                 |
+| Priority | As a …​                                    | I want to …​                     | So that I can…​                                     |
```
Comment (KevinEyo1) [inline docs/DeveloperGuide.md:273]: Perhaps change this to "I can" to allign with user stories?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -270,58 +270,115 @@ _{Explain here how the data archiving feature will be implemented}_
...
+
+1a. Ultron detects an error in the entered data.
+    1a1. Ultron informs the user that the data is invalid
+    Use case ends
+
+**Use case:  UC 3 - Editing an opening**
+
+**MSS**
+
+1. User enters a command to edit an opening
```
Comment (KevinEyo1) [inline docs/DeveloperGuide.md:323]: Perhaps change lines like this (applies to other use cases) to "User enters opening with the details to change it."?
Can avoid using "command"

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -270,58 +270,115 @@ _{Explain here how the data archiving feature will be implemented}_
...
-* 2a. The list is empty.
+1a. Ultron detects an error in the entered data.
+    1a1. Ultron informs the user that the data is invalid
+    Use case ends
+
+**Use case:  UC 2 - Deleting an opening**
+
+**MSS**
+
+1. User enters a command to delete an opening
```
Comment (KevinEyo1) [inline docs/DeveloperGuide.md:308]: Perhaps add full stops to all instructions?

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -270,58 +270,115 @@ _{Explain here how the data archiving feature will be implemented}_
...
+2. The app should be fast and respond within 100ms when users input commands.
+3. The app should start up and load the list from a saved file within 1 second of opening.
 
 *{More to be added}*
 
 ### Glossary
 
 * **Mainstream OS**: Windows, Linux, Unix, OS-X
-* **Private contact detail**: A contact detail that is not meant to be shared with others
+* **Opening**: Item representing an internship opening
```
Comment (KevinEyo1) [inline docs/DeveloperGuide.md:380]: Perhaps could change to "Item containing all relevant details for a particular internship application"

**Review by KevinEyo1 — COMMENTED** [review]
LGTM, few nits to fix

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (Lanreath) [reply docs/DeveloperGuide.md:323]: User selects opening and includes updated details

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -270,58 +270,115 @@ _{Explain here how the data archiving feature will be implemented}_
...
+| -------- | ------------------------------------------ | ------------------------------ |-----------------------------------------------------|
+| `* * *`  | potential user exploring the app           | see the app populated with sample openings         | easily see how the app will look when it is in use. |
+| `* * *`  | user                                       | add an opening regarding the internship I am interested in               | keep note of it                                     |
+| `* * *`  | user                                       | delete an opening      | remove unwanted openings                            |
+| `* * *`  | user                                       | Be able to see all applications          | track my progress                                   |
+| `* *`    | user                                       | see submission deadlines   | check them                                          |
+| `*`      | user | be able to see the total number of accepted applications           | celebrate my success                              |
+| '*' | user | see all the outcomes of my application | properly assess my options |
```
Comment (Lanreath) [inline docs/DeveloperGuide.md:281]: More user stories should be added depending on what we plan to implement

**Review by Lanreath — APPROVED** [review]

**Pull request merged**